### ISSUE TITLE
panic runtime error using AWS S3 source with --s3-download-url option

### ISSUE THREAD

**Issue by mohamedamjad** [rclone/rclone#8126]
#### What is the problem you are having with rclone?

panic: runtime error: invalid memory address or nil pointer dereference

#### What is your rclone version (output from `rclone version`)

```
rclone v1.68.1
- os/version: darwin 14.6.1 (64 bit)
- os/kernel: 23.6.0 (x86_64)
- os/type: darwin
- os/arch: amd64
- go/version: go1.23.1
- go/linking: dynamic
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

```
macOS 14.6.1
```

#### Which cloud storage system are you using? (e.g. Google Drive)
AWS S3

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

```
rclone copy cloudf:/xxxxxxxx/xxxxxx/xxx ./ --s3-download-url https://xxxxxx.cloudfront.net/ -vv
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2024/10/11 17:40:15 DEBUG : rclone: Version "v1.68.1" starting with parameters ["rclone" "copy" "cloudf:/xxx/xxx/x" "./" "--s3-download-url" "https://xxxxxx.cloudfront.net/" "-vv"]
2024/10/11 17:40:15 DEBUG : Creating backend with remote "cloudf:/xxxxxx/xxxxxx/x"
2024/10/11 17:40:15 DEBUG : Using config file from "/xxxxxxxxxxxxxxx"
2024/10/11 17:40:15 DEBUG : cloudf: detected overridden config - adding "{CH1CJ}" suffix to name
2024/10/11 17:40:15 DEBUG : fs cache: renaming cache item "cloudf:/xx-xx/xx/xxxx" to be canonical "cloudf{CH1CJ}:xxxx-xxxxx/xxxxxx/xxxxx"
2024/10/11 17:40:15 DEBUG : Creating backend with remote "./"
2024/10/11 17:40:15 DEBUG : fs cache: renaming cache item "./" to be canonical "/xxxxx/xxxxxx/xxxxx/xxxxxxxx"
2024/10/11 17:40:16 DEBUG : xxxxx/xxxxxxxxx/xxxxxxxs.xxxxxxx: Need to transfer - File not found at Destination
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10539a21e]

goroutine 88 [running]:
github.com/rclone/rclone/backend/s3.(*Object).downloadFromURL(0xc0009003f0, {0x106626b30, 0xc0000e8870}, {0xc0000622ac, 0x50}, {0xc0000ce520, 0x1, 0x1})
	github.com/rclone/rclone/backend/s3/s3.go:5749 +0xa7e
github.com/rclone/rclone/backend/s3.(*Object).Open(0xc0009003f0, {0x106626b30, 0xc0000e8870}, {0xc0000ce520, 0x1, 0x1})
	github.com/rclone/rclone/backend/s3/s3.go:5760 +0x387
github.com/rclone/rclone/fs/operations.(*ReOpen).open(0xc000932380)
	github.com/rclone/rclone/fs/operations/reopen.go:147 +0x109
github.com/rclone/rclone/fs/operations.NewReOpen({0x106626b30, 0xc0000e8870}, {0x10663e630, 0xc0009003f0}, 0xa, {0xc0000ce510, 0x1, 0x1})
	github.com/rclone/rclone/fs/operations/reopen.go:102 +0x50b
github.com/rclone/rclone/fs/operations.Open({0x106626b30, 0xc0000e8870}, {0x10663e630, 0xc0009003f0}, {0xc0000ce510, 0x1, 0x1})
	github.com/rclone/rclone/fs/operations/reopen.go:124 +0x6b
github.com/rclone/rclone/fs/operations.(*copy).manualCopy(0xc0009901b0, {0x106626b30, 0xc0000e8870})
	github.com/rclone/rclone/fs/operations/copy.go:261 +0x4dc
github.com/rclone/rclone/fs/operations.(*copy).copy(0xc0009901b0, {0x106626b30, 0xc0000e8870})
	github.com/rclone/rclone/fs/operations/copy.go:310 +0x176
github.com/rclone/rclone/fs/operations.Copy({0x106626b30, 0xc0000e8870}, {0x10663d750, 0xc0008bc8c0}, {0x0, 0x0}, {0xc000b1e73c, 0x44}, {0x10663e630, 0xc0009003f0})
	github.com/rclone/rclone/fs/operations/copy.go:412 +0x493
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairCopyOrMove(0xc0004fea08, {0x106626b30, 0xc0000e8870}, 0xc0000e8690, {0x10663d750, 0xc0008bc8c0}, 0x32, 0x0?)
	github.com/rclone/rclone/fs/sync/sync.go:501 +0x1fe
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startTransfers in goroutine 1
	github.com/rclone/rclone/fs/sync/sync.go:531 +0x6b

```

**Comment by darthShadow** [comment]
I believe this has already been fixed in the beta (https://github.com/rclone/rclone/commit/1ca3f1267297b1e3ac45df2e9e240570bff62c84). Please try it out and report back if it's still having issues.

**Comment by helixzz** [comment]
I'm also running into this bug in v1.68.0 (DEB package).
Currently I worked around the problem by downgrading to v1.67.0.
Looking forward to see a new DEB package release since we're not planning to compile ourselves.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x18f20be]

goroutine 545 [running]:
github.com/rclone/rclone/backend/s3.(*Object).downloadFromURL(0xc00198a2d0, {0x2a61df0, 0xc0032440a0}, {0xc0033cc016, 0x44}, {0xc003382230, 0x1, 0x1})
        github.com/rclone/rclone/backend/s3/s3.go:5748 +0xa7e
github.com/rclone/rclone/backend/s3.(*Object).Open(0xc00198a2d0, {0x2a61df0, 0xc0032440a0}, {0xc003382230, 0x1, 0x1})
        github.com/rclone/rclone/backend/s3/s3.go:5759 +0x387
github.com/rclone/rclone/fs/operations.(*ReOpen).open(0xc0033982a0)
        github.com/rclone/rclone/fs/operations/reopen.go:147 +0x109
github.com/rclone/rclone/fs/operations.NewReOpen({0x2a61df0, 0xc0032440a0}, {0x2a798b0, 0xc00198a2d0}, 0xa, {0xc003382220, 0x1, 0x1})
        github.com/rclone/rclone/fs/operations/reopen.go:102 +0x50b
github.com/rclone/rclone/fs/operations.Open({0x2a61df0, 0xc0032440a0}, {0x2a798b0, 0xc00198a2d0}, {0xc003382220, 0x1, 0x1})
        github.com/rclone/rclone/fs/operations/reopen.go:124 +0x6b
github.com/rclone/rclone/fs/operations.(*multiThreadCopyState).copyChunk(0xc0032440f0, {0x2a61df0, 0xc0032440a0}, 0x0, {0x2a5ba68, 0xc003202f60})
        github.com/rclone/rclone/fs/operations/multithread.go:84 +0x2df
github.com/rclone/rclone/fs/operations.multiThreadCopy.func2()
        github.com/rclone/rclone/fs/operations/multithread.go:228 +0x2b
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 126
        golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x96
```

**Comment by ncw** [comment]
@helixzz you can find binaries including debs at [the beta download site](https://beta.rclone.org/)
 
- [ rclone-beta-latest-linux-mipsle.deb](https://beta.rclone.org/rclone-beta-latest-linux-mipsle.deb)
- [ rclone-beta-latest-linux-mips.deb](https://beta.rclone.org/rclone-beta-latest-linux-mips.deb)
- [ rclone-beta-latest-linux-arm64.deb](https://beta.rclone.org/rclone-beta-latest-linux-arm64.deb)
- [ rclone-beta-latest-linux-arm.deb](https://beta.rclone.org/rclone-beta-latest-linux-arm.deb)
- [ rclone-beta-latest-linux-arm-v7.deb](https://beta.rclone.org/rclone-beta-latest-linux-arm-v7.deb)
- [ rclone-beta-latest-linux-arm-v6.deb](https://beta.rclone.org/rclone-beta-latest-linux-arm-v6.deb)
- [ rclone-beta-latest-linux-amd64.deb](https://beta.rclone.org/rclone-beta-latest-linux-amd64.deb)
- [ rclone-beta-latest-linux-386.deb](https://beta.rclone.org/rclone-beta-latest-linux-386.deb)

I'm going to close this as it was fixed in 1ca3f1267297b1e3ac45df2e9e240570bff62c84
